src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenChatHandler.java

```diff
@@ -47,78 +44,65 @@ public void notifyDeath(final DamageSource damageSource)
         if (citizen.getCitizenColonyHandler().getColony() != null && citizen.getCitizenData() != null)
         {
             final IJob<?> job = citizen.getCitizenJobHandler().getColonyJob();
+            TranslationTextComponent contentComponent;
             if (job != null)
             {
-                final ITextComponent component = new TranslationTextComponent(
-                        "block.blockhuttownhall.messageworkerdead",
-                        new TranslationTextComponent(job.getJobRegistryEntry().getTranslationKey()),
-                        citizen.getCitizenData().getName(),
-                        (int) citizen.getX(), (int) citizen.getY(),
-                        (int) citizen.getZ(), new TranslationTextComponent(damageSource.msgId));
-                LanguageHandler.sendPlayersMessage(citizen.getCitizenColonyHandler().getColony().getImportantMessageEntityPlayers(), "", component);
+                contentComponent = new TranslationTextComponent(
+                  TranslationConstants.WORKER_DIED,
+                  new TranslationTextComponent(job.getJobRegistryEntry().getTranslationKey()),
+                  citizen.getCitizenData().getName(),
+                  Math.round(citizen.getX()),
+                  Math.round(citizen.getY()),
+                  Math.round(citizen.getZ()),
+                  new TranslationTextComponent(damageSource.msgId));
             }
             else
             {
-                LanguageHandler.sendPlayersMessage(
-                        citizen.getCitizenColonyHandler().getColony().getImportantMessageEntityPlayers(), "",
-                        new TranslationTextComponent("block.blockhuttownhall.messagecolonistdead",
-                                citizen.getCitizenData().getName(), (int) citizen.getX(), (int) citizen.getY(),
-                                (int) citizen.getZ(), new TranslationTextComponent(damageSource.msgId)));
+                contentComponent = new TranslationTextComponent(
+                  TranslationConstants.COLONIST_DIED,
+                  citizen.getCitizenData().getName(),
+                  Math.round(citizen.getX()),
+                  Math.round(citizen.getY()),
+                  Math.round(citizen.getZ()),
+                  new TranslationTextComponent(damageSource.msgId));
             }
+
+            MessageUtils.format(contentComponent)
+              .with(TextFormatting.RED)
+              .sendTo(citizen.getCitizenColonyHandler().getColony()).forManagers();
         }
     }
 
     @Override
     public void sendLocalizedChat(final String keyIn, final Object... msg)
```

Comment:
Would make sense for me to remove this method

Reply:
Whilst I agree with the fact that the method technically is only used once, and therefore shouldn't be needed, I don't think cramming everything together in one method isn't great practice either.
This class already exists to handle messages related to citizens, so rather keep it and expand on it in the future, moving more messaging to this class instead.